Branch-SDK/Branch-SDK/BNCPreferenceHelper.m

```diff
@@ -181,15 +181,7 @@ + (NSInteger)getRetryCount {
 }
 
 + (NSString *)getAppKey {
-    NSString *ret = [[[NSBundle mainBundle] infoDictionary] objectForKey:KEY_APP_KEY];
-    if (!ret || ret.length == 0) {
-        // for backward compatibility
-        ret = [BNCPreferenceHelper readStringFromDefaults:KEY_APP_KEY];
```

Comment:
Way before the introduction of Branch key, when we used to store app key in defaults, but moved it to plist. It was at that time this "backward compatibility" line of code was added, I think.
But, I agree on your point. We should disable for both old and new users in case the Branch key is missing.